comictaggerlib/taggerwindow.py

```diff
@@ -1219,24 +1221,25 @@ def commit_metadata(self) -> None:
                 self.update_menus()
             self.fileSelectionList.update_current_row()
 
-            self.metadata = self.comic_archive.read_metadata(self.load_data_style)
+            self.metadata = GenericMetadata()
+            for style in reversed(self.load_data_styles):
+                self.metadata.overlay(self.comic_archive.read_metadata(style))
             self.update_ui_for_archive()
         else:
             QtWidgets.QMessageBox.information(self, "Whoops!", "No data to commit!")
 
-    def set_load_data_style(self, s: str) -> None:
+    def set_load_data_style(self) -> None:
+        # TODO Should the message be changed? "manually entered data will be lost" Better desc for all 3 options?
+        # Save, Discard and Cancel - Cancel will keep the data in the form AND change the style. Explain each one?
         if self.dirty_flag_verification(
-            "Change Tag Read Style", "If you change read tag style now, data in the form will be lost.  Are you sure?"
+            "Change Tag Read Style",
+            "If you change read tag style(s) now, data in the form will be lost.  Are you sure?",
         ):
-            self.load_data_style = self.cbLoadDataStyle.itemData(s)
-            self.config[0].internal__load_data_style = self.load_data_style
+            self.load_data_styles = self.cbLoadDataStyle.currentData()
+            self.config[0].internal__load_data_style = self.load_data_styles
             self.update_menus()
             if self.comic_archive is not None:
                 self.load_archive(self.comic_archive)
-        else:
-            self.cbLoadDataStyle.currentIndexChanged.disconnect(self.set_load_data_style)
-            self.adjust_load_style_combo()
-            self.cbLoadDataStyle.currentIndexChanged.connect(self.set_load_data_style)
```

Comment:
Is the cancel option supposed to work as I've found: "Cancel will keep the data in the form AND change the style."? If so, making the message clearer might be an idea.

Reply:
Ah, I see it now. I thought because `currentIndexChanged` wasn't needed to be changed back it could go. Update coming.